client/lib/plugins/sanitize-section-content.js

```diff
@@ -0,0 +1,187 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import validUrl from 'valid-url';
+
+const isAllowedTag = tagName => {
+	switch ( tagName ) {
+		case '#text':
+		case 'a':
+		case 'b':
+		case 'blockquote':
+		case 'code':
+		case 'div':
+		case 'em':
+		case 'h1':
+		case 'h2':
+		case 'h3':
+		case 'h4':
+		case 'h5':
+		case 'h6':
+		case 'i':
+		case 'img':
+		case 'li':
+		case 'ol':
+		case 'p':
+		case 'span':
+		case 'strong':
+		case 'ul':
+			return true;
+		default:
+			return false;
+	}
+};
+
+const isAllowedAttr = ( tagName, attrName ) => {
+	if ( 'a' === tagName && 'href' === attrName ) {
+		return true;
+	}
+
+	if ( 'iframe' === tagName ) {
+		switch ( attrName ) {
+			case 'class':
+			case 'type':
+			case 'height':
+			case 'width':
+			case 'src':
+				return true;
+			default:
+				return false;
+		}
+	}
+
+	if ( 'img' === tagName && ( 'alt' === attrName || 'src' === attrName ) ) {
+		return true;
+	}
+
+	return false;
+};
+
+const isValidYoutubeEmbed = node => {
+	if ( node.nodeName.toLowerCase() !== 'iframe' ) {
+		return false;
+	}
+
+	if ( node.getAttribute( 'class' ) !== 'youtube-player' ) {
+		return false;
+	}
+
+	if ( node.getAttribute( 'type' ) !== 'text/html' ) {
+		return false;
+	}
+
+	const link = document.createElement( 'a' );
+	link.href = node.getAttribute( 'src' );
+
+	return (
+		validUrl.isWebUri( node.getAttribute( 'src' ) ) &&
+		( link.hostname === 'youtube.com' || link.hostname === 'www.youtube.com' )
+	);
+};
+
+/**
+ * Sanitizes input HTML for security and styling
+ *
+ * @param {String} content unverified HTML
+ * @returns {string} sanitized HTML
+ */
+export const sanitizeSectionContent = content => {
+	const parser = new DOMParser();
+	const doc = parser.parseFromString( content, 'text/html' );
+
+	// this will let us visit every single DOM node programmatically
+	const walker = doc.createTreeWalker(
+		doc.body,
+		NodeFilter.SHOW_ALL,
```

Comment:
do we need any of the params after the first one? Is whats currently written equivalent to:
```js
const walker = doc.createTreeWalker( doc.body );
```

Reply:
Defaults accepted in 19e11d6c6